index.js

```diff
@@ -0,0 +1,82 @@
+const Base = require('bfx-facs-base')
+const crypto = require('crypto')
+const fs = require('fs/promises')
+const path = require('path')
+const TTLCache = require('@isaacs/ttlcache')
+
+class TTLCacheFs extends Base {
+  constructor (caller, opts = {}, ctx) {
+    super(caller, opts, ctx)
+
+    this.name = 'ttlcache_fs'
+    this._hasConf = false
+
+    this._cacheDirectory = opts.directory ?? './cache/data'
+
+    if (opts?.deleteCacheDirOnStart ?? true) {
+      // Clear cache directory on start
+      fs.rm(this._cacheDirectory, {
+        force: true,
+        recursive: true
+      }).catch(err => console.trace(`Could not clear cache directory ${this._cacheDirectory}`, err))
+    }
+
+    const cacheOpts = {
+      max: 1000,
+      ttl: 3600 * 1000,
+      dispose: (value, key) => {
```

Comment:
it seems that the main reason for using ttlcache is for this dispose method, is there not a custom cache library that stores cache in the fs rather than memory? I found this but not tested - https://www.npmjs.com/package/cacache

Reply:
I was surprised as well. I'd love to use an fs cache with ttl but could not find a working and supported module.
The `cacache` does not offer ttl so we would have to continue using `TTLCache` either way.

We could replace the fs cache implementation with `cacache`, but I'm not sure if it's worth it since our implementation is very straightforward.